include/RAJA/config.hpp.in

```diff
@@ -195,24 +195,24 @@ const int DATA_ALIGN = @RAJA_DATA_ALIGN@;
 #define RAJA_BOUNDS_CHECK_constexpr
 
 #if !defined(NDEBUG)
-#define RAJA_ASSERT(EXP) assert( (EXP) )
+#define RAJA_EXIT_KERNEL assert( 0 );
```

Comment:
Don't all these guys need to take an expression to make the code compile and work like they did before the name change?

Reply:
The expression was only needed for assert. When assert is not defined, abort(), or asm ("trap;") is used and they don't require an expression. Since we want to the kernel to exit we can just use 0 for the assert argument.

Reply:
Hmmm. I guess I'm confused then. Shouldn't these trigger on some boolean expression?

Reply:
I think the intent is to have the outer if statement perform the check, and RAJA_EXIT_KERNEL is a simple host-defined exit based on the result of the outer check. This removes the need for RAJA_EXIT_KERNEL to accept another lengthy boolean expression.

Reply:
Doesn't assert print a stringified version of its argument? Knowing what failed can be useful sometimes.

Reply:
I would call this abort instead of exit.

Reply:
I would be fine with having both assert and abort, abort could just call assert(0);

Reply:
@MrBurmark , I would prefer abort as well; however,  if users build with CMAKE in release mode NDEBUG is defined which makes assert a no-op.

Reply:
@artv3 adding a boolean expression or string argument that can be printed out would be very useful as @MrBurmark suggested.

Reply:
> Hmmm. I guess I'm confused then. Shouldn't these trigger on some boolean expression?

The boolean expression is inside the bounds check routine: https://github.com/LLNL/RAJA/pull/789/files#diff-384a1962d3b77bda64b52c37bd9a2ca6R240 . This lets us add the CUDA friendly print statement which gives us the index which was out of bounds.

Reply:
That's fine, we're just naming our construct. Then we can implement it to do the behavior regardless of the setting, though the niceness of the output may vary based on the setting.

Reply:
> I would call this abort instead of exit.

Oh do you mean RAJA_ABORT_KERNEL rather than RAJA_EXIT_KERNEL?

Reply:
Ok, if we're already taking care of the print separately I see your point. Is RAJA_ASSERT strictly internal or are we going to expose this to users?

Reply:
> 
> 
> > I would call this abort instead of exit.
> 
> Oh do you mean RAJA_ABORT_KERNEL rather than RAJA_EXIT_KERNEL?

Yes

Reply:
> Ok, if we're already taking care of the print separately I see your point. Is RAJA_ASSERT strictly internal or are we going to expose this to users?

I was thinking it could be internal? I haven't thought of exposing it to users. Do we want to do that?

Reply:
> I was thinking it could be internal? I haven't thought of exposing it to users. Do we want to do that?

Let's not worry about that then.

Reply:
On a different note, I wasn't able to build with HIP. If this is okay, we can merge this and I can make a follow up PR once we update host configs.

Reply:
The corona script seems to be out of date with actual system changes on corona. I was not able to build other tests either, and can look in to fixing the script after I update the atomic forall tests and reduction unit tests.